internal/grpc/services/userprovider/userprovider.go

```diff
@@ -59,7 +71,38 @@ func parseConfig(m map[string]interface{}) (*config, error) {
 	return c, nil
 }
 
-func getDriver(c *config) (user.Manager, error) {
+// getDriverPlugin fetches the runtime driver from the plugins package
+func getDriverPlugin(c *config) (user.Manager, error) {
+	sym, err := c.load()
+	if err != nil {
+		return nil, err
+	}
+
+	// assert the loaded plugin into required interface
+	manager, ok := sym.(user.Manager)
+	if !ok {
+		return nil, fmt.Errorf("could not assert the loaded plugin")
+	}
+
+	basename := filepath.Base(c.Driver)
+	pluginConfig := strings.TrimSuffix(basename, filepath.Ext(basename))
```

Comment:
Again, I need to do this because, whenever a user provides configure like:
```
[grpc.services.userprovider]
driver = /home/path/to/json.go

[grpc.services.userprovider.json]
users = users.demo.json
```
so I need to strip off, `.go` from the driver field to access to users field!

Reply:
Pointing to the package should take care of that